packages/xstate-solid/test/useMachine.test.tsx

```diff
@@ -0,0 +1,1612 @@
+        }
+      }
+    });
+
+    const App = () => {
+      const [state, send] = useMachine(machine);
+      const [toJson, setToJson] = createSignal(state.toJSON());
+      createEffect(
+        on(
+          () => state.value,
+          () => {
+            setToJson(state.toJSON());
+          }
+        )
+      );
+      return (
+        <div>
+          <button
+            data-testid="transition-button"
+            onclick={() => send({ type: 'TRANSITION' })}
+          />
+          <div data-testid="to-json">{toJson().value.toString()}</div>
+        </div>
+      );
+    };
+
+    render(() => <App />);
+    const toJsonEl = screen.getByTestId('to-json');
+    const transitionBtn = screen.getByTestId('transition-button');
+
+    // Green
+    expect(toJsonEl.textContent).toEqual('green');
+    transitionBtn.click();
+
+    // Yellow
+    expect(toJsonEl.textContent).toEqual('yellow');
+    transitionBtn.click();
+
+    // Red
+    expect(toJsonEl.textContent).toEqual('red');
+    transitionBtn.click();
+
+    // Green
+    expect(toJsonEl.textContent).toEqual('green');
+  });
+
+  it('should be reactive to hasTag method calls', () => {
+    const machine = createMachine({
+      initial: 'green',
+      states: {
+        green: {
+          tags: 'go',
+          on: {
+            TRANSITION: 'yellow'
+          }
+        },
+        yellow: {
+          tags: 'go',
+          on: {
+            TRANSITION: 'red'
+          }
+        },
+        red: {
+          tags: ['stop', 'other'],
+          on: {
+            TRANSITION: 'green'
+          }
+        }
+      }
+    });
+
+    const App = () => {
+      const [state, send] = useMachine(machine);
+      const [canGo, setCanGo] = createSignal(state.hasTag('go'));
+      createEffect(() => {
+        setCanGo(state.hasTag('go'));
+      });
+      return (
+        <div>
+          <button
+            data-testid="transition-button"
+            onclick={() => send({ type: 'TRANSITION' })}
+          />
+          <div data-testid="can-go">{canGo().toString()}</div>
+          <div data-testid="stop">{state.hasTag('stop').toString()}</div>
+        </div>
+      );
+    };
+
+    render(() => <App />);
+    const canGoEl = screen.getByTestId('can-go');
+    const stopEl = screen.getByTestId('stop');
+    const transitionBtn = screen.getByTestId('transition-button');
+
+    // Green
+    expect(canGoEl.textContent).toEqual('true');
+    expect(stopEl.textContent).toEqual('false');
+    transitionBtn.click();
+
+    // Yellow
+    expect(canGoEl.textContent).toEqual('true');
+    expect(stopEl.textContent).toEqual('false');
+    transitionBtn.click();
+
+    // Red
+    expect(canGoEl.textContent).toEqual('false');
+    expect(stopEl.textContent).toEqual('true');
+    transitionBtn.click();
+
+    // Green
+    expect(canGoEl.textContent).toEqual('true');
+    expect(stopEl.textContent).toEqual('false');
+  });
+
+  it('should be reactive to can method calls', () => {
```

Comment:
Should `state.can(FOO)` be "embeddable" and reactive in the template part? or should this always be used like this there: `() => state.can(FOO)`?

Reply:
It's important to be embeddable in the template for conditional rendering like with: `<Show when={state.can(FOO)}>`. We can't make it not reactive only in the template; it's all in on reactivity for a value unless a user manually `untrack`s it.

Reply:
Right, i want it to be reactive in tbe template - just wasnt sure what’s the preferred way to access this value in the template for it to be reactive etc

Reply:
Coming in late as trying to rejig existing old machines into SolidStart and only half know what's going on - currently using `state().can({ type: 'FOO' })` as my syntax, `state` being reactive signal/function, with type specified in the `.can()` bit.